### ISSUE TITLE
Add nested comments integration

### ISSUE THREAD

**Issue by ljades** [AlexPadron/comment_formatter#9]
It would be really cool to add support for nested comment blocks. For example:

```
# Here is a list:
#  1. Item A.
#  2. Item B.
```

If possible, adding a recursive element to the nesting would also be amazing.

```
# The following should happen in this method:
#  1. Non-nested comments should work.
#  2. Nested comments should work.
#      -  This includes recursive nesting for nested sub-blocks.
```

This would also hopefully involve formatting different levels of spacing to align properly. For example, set a standard for number of spaces in an indentation given a nesting level (probably 4, including the numbering/bullet points.

**Comment by AlexPadron** [comment]
agreed we should add this - we probably need some indication of indentation, possibly a leading space, like:

```python
# This comment
#  1. is in a separate block from this one
#  2. but these two are together
```

The difficulty is telling the difference between a `1.` that got reformatted to be at the start of a line, and indentation that is there on purpose

**Comment by ljades** [comment]
Just edited to add that change.
